code/script.js

```diff
@@ -237,107 +241,189 @@ const generateBoard = () => {
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  setSecret()
+  generateBoard()
+  countClicks = 0;
 }
 
-// setting the currentQuestion object when you select something in the dropdown
-const selectQuestion = () => {
+// Compares your question with the secret person
+// and stores the value in currentQuestion
+const selectQuestion = (selected) => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-
-  if (category === 'hair color') {
+  if (category === 'other') {
+    currentQuestion = {
+      attribute: 'smoker',
+      value: true,
+      category: category,
+    }
+  } else if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
+      value: selected,
       category: category,
     }
   } else if (category === 'eye color') {
-    // Set this up your self
-  } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
+      attribute: 'eyeColor',
+      value: selected,
       category: category,
     }
-  } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+  } else if (category === 'accessories') {
+    if (selected === 'hat') {
+      currentQuestion = {
+        attribute: 'hat',
+        value: true,
+        category: category,
+      }
+    } else if (selected === 'glasses') {
+      currentQuestion = {
+        attribute: 'glasses',
+        value: true,
+        category: category,
+      }
+    }
   }
 }
 
-// This function should be invoked when you click on 'Find Out'.
+// Checks if your guess matches the secret person
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  const secretValue = secret[currentQuestion.attribute]
+  if (secretValue === currentQuestion.value) {
+    keep = true
+    filterCharacters(true)
+  } else {
+    keep = false
+    filterCharacters(false)
+  }
 }
 
-// It'll filter the characters array and redraw the game board.
+// Filters charachters depending on your guess and redraws the board
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
+  const value = currentQuestion.value;
+  const attribute = currentQuestion.attribute;
+
+  if (attribute === 'glasses') {
     if (keep) {
```

Comment:
This `keep` is not a global `keep` but the one which is existing only inside of this function- created in line 317.